### PR TITLE
Add support for reading long, binary double and auto increment field values.

### PR DESCRIPTION
**Pull request by PatrickHofman**
This commit holds a fix for issue #29.

### CODE REVIEW

**Comment by monty241** [comment]
Good remark; crossed the mind and zapped away unfortunately.

File: DotNetDBF/DBFReader.cs [hunk 1]
```diff
@@ -462,10 +462,59 @@ internal object[] NextRecord(IEnumerable<int> selectIndexes, IList<int> sortedIn
...
-                            _dataInputStream.ReadBytes(_header.FieldArray[i].FieldLength);
-                            recordObjects[i] = DBNull.Value;
-                            break;
+                            {
+                                byte[] data = _dataInputStream.ReadBytes(_header.FieldArray[i].FieldLength);
+
+                                recordObjects[i] = data != null ? (object)data : DBNull.Value;
+
+                                break;
+                            }
```
Comment (monty241) [inline DotNetDBF/DBFReader.cs]: Keep this one.

**Comment by PatrickHofman** [comment]
@jbtule Thanks for your reply. Since this is Visual FoxPro, I am not sure what the correct handling would be, since it is not documented any where. It seems to be a little endian double precision IEEE 754 float. There are two possible solutions I can come up with:

- Make a check if little endian is used. If so, use BitConverter. If not, either raise an exception, or reorder the bytes and call BitConverter.
- Remove the handling for Binary entirely. Keep the part that returns a byte array as default, so the implementer can handle this themselves (like in our case, we know some characteristics which makes it saves to assume we can read it a certain way).

What do you propose?

**Comment by jbtule** [comment]
I think the byte array default make a lot of sense.

**Pull request merged**